### PR TITLE
Implement Podcast Data Enrichment Pipeline Using OpenAI API

### PR DESCRIPTION
**Pull request by yanirmr**
### Description
This pull request introduces a comprehensive pipeline for enriching podcast episode data using the OpenAI API. Our primary goal is to extract and add insightful metadata to our existing podcast episode dataset, which includes fields like title, number, duration, and description. The enrichment involves adding information about episode participants, their genders, and general topics, obtained via AI-driven analysis.

### Key Components
**RSS File Parsing (`rss_parser.py`):**
- Parses podcast episode information from an RSS feed in XML format.
- Extracts key details such as title, episode number, duration, and description.

**Interacts with the OpenAI API to analyze podcast descriptions (`api_integration.py`)**
- Extracts structured information like participant names, genders, and episode topics.
- Formats API requests and parses responses into a JSON structure for consistent processing.

**Data Aggregation (`data_aggregator.py`) - Combines the parsed RSS data with API results.**
- Enriches the original dataset with new columns based on the API's response.
- Offers functionality to process either the full dataset or a sampled subset.

### Implementation Details
- The pipeline is designed with a focus on modularity, allowing each component to function independently yet integrate seamlessly.
- Special attention has been given to language processing, particularly for handling Hebrew in podcast descriptions.
- The process concludes with exporting the enriched dataset, appending '_enriched' to the original filename, which provides a clear lineage of the data transformation.

### Usage
The pipeline is primarily intended for internal use to enhance our podcast dataset. It can be triggered via the `data_agregator.py` script, which is configured to process a random sample of episodes for initial testing and can be easily adjusted for full-scale processing.

### CODE REVIEW

**Review by yairl — COMMENTED** [review]
I think my main concern is that it's structured as a mini-package instead of a single file.
My intuition - could be wrong - is that this is a case where extra modularity is a downside.

Example:

Once you remove documentation and examples, rss_parser.py is <20 lines.
Same is mostly true for the other files.

I would suggest dropping the csv and xml files (XML should be loaded dynamically, CSV should never be generated) and treat this as one step in our processing pipeline.

Input: XML
Output: JSON per XML

We can then change our remaining code (process.py, transcribe.py) to augment that JSON as necessary with VAD and transcription information.

A single file for this task would be small, self-contained, and IMO easier to understand and handle.

File: rss_parser/api_integration.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    prompt = f"Analyze the following podcast episode data: {text}\n"
+    if context_info:
+        prompt += f"Context: {json.dumps(context_info)}\n"
+    prompt += (
+        "Extract the participant names from the title, their genders (male, female, unknown), "
+        "and the main topics. Format the response as a JSON object with the keys "
+        "'participants', 'genders', and 'topics'. For example: \n"
+        "{\n"
+        "  \"participants\": [\"Alice\", \"Bob\"],\n"
```
[the remark this answers is not in the document]
Comment (yanirmr) [reply rss_parser/api_integration.py]: This is the only comment I'm not sure I get.

File: rss_parser/api_integration.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    :return: Extracted information as a dictionary.
+    """
+    # Remove the extra characters
+    json_str = response_text.replace('```json\n', '').replace('\n```', '').strip()
+
+    # Parse the string as JSON
+    try:
+        return json.loads(json_str)
+    except json.JSONDecodeError as e:
+        print(f"Error parsing JSON: {e}")
```
[the remark this answers is not in the document]
Comment (yanirmr) [reply rss_parser/api_integration.py]: removed.

File: rss_parser.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                                      model_name: str = "gpt-4-1106-preview") -> str:
+    """
+    Sends a request to the OpenAI API to analyze podcast episode data and extract specific information.
+
+    :param podcast_text: The text of the podcast episode for analysis.
+    :param podcast_context: Additional context information for the API request.
+    :param model_name: The name of the OpenAI model to be used. Default is 'gpt-4-1106-preview'.
+    :return: The API response as a JSON string.
+    """
+    api_key = os.getenv('OPENAI_API_KEY')
```
Comment (yairl) [inline rss_parser.py:41]: Good to add this to .env.example

File: rss_parser.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+def extract_information_from_response(response_text: str) -> dict:
+    """
+    Extracts information from the OpenAI API response_text.
+
+    :param response_text: The response_text from the OpenAI API.
+    :return: Extracted information as a dictionary.
+    """
+    # Remove the extra characters
+    json_str = response_text.replace('```json\n', '').replace('\n```', '').strip()
```
Comment (yairl) [inline rss_parser.py:76]: Using XML tags works better.
OK for this commit, I will explain separately offline.

File: rss_parser.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    for episode in episodes:
+        podcast_text = f"{episode['title']} {episode['description']}"
+        content_data = extract_podcast_data_from_content(podcast_text)
+        extracted_info = extract_information_from_response(content_data)
+        episode.update(extracted_info)
+
+    return json.dumps(episodes, indent=4)
+
+
+if __name__ == "__main__":
```
Comment (yairl) [inline rss_parser.py:99]: Would be good to use argparse here.
Not critical for this commit, but good to add separately.

Ideally, this should iterate over all feed.xml files (similar to how --audio-dir works in other scripts).

**Review by yairl — APPROVED** [review]

**Pull request merged**